File: core/src/main/java/com/mcreater/amclcore/command/script/ShellScriptProvider.java
package com.mcreater.amclcore.command.script;

import com.mcreater.amclcore.command.CommandArg;

import java.util.List;
import java.util.stream.Collectors;

public class ShellScriptProvider implements ScriptProvider {
    public static final ShellScriptProvider INSTANCE = new ShellScriptProvider();

    private ShellScriptProvider() {
    }

    public String toScript(List<CommandArg> args) {
        return args.stream()
                .map(CommandArg::toString)
                .map(s -> "\"" + s + "\"")
                .collect(Collectors.joining(" "));
    }

    public String getFileExtension() {
        return ".sh";
    }
}

File: core/src/main/java/com/mcreater/amclcore/command/script/ShellScriptProvider.kt
package com.mcreater.amclcore.command.script

import com.mcreater.amclcore.command.CommandArg
import java.util.stream.Collectors


class ShellScriptProvider private constructor() : ScriptProvider {
    override fun toScript(args: List<CommandArg?>?): String? {
        return args?.stream()
            ?.map { it.toString() }
            ?.map { "\"" + it + "\"" }
            ?.collect(Collectors.joining(" "))
    }

    override val fileExtension: String
        get() = ".sh"

    companion object {
        val INSTANCE = ShellScriptProvider()
    }
}

